core/store/orm/orm_test.go

```diff
@@ -61,11 +61,11 @@ func TestORM_Unscoped(t *testing.T) {
 
 	orm := store.ORM
 	job := cltest.NewJob()
-	err := orm.RawDB(func(db *gorm.DB) error {
+	err := orm.RawDBWithAdvisoryLock(func(db *gorm.DB) error {
 		require.NoError(t, orm.CreateJob(&job))
 		require.NoError(t, db.Delete(&job).Error)
 		require.Error(t, db.First(&job).Error)
-		err := store.ORM.Unscoped().RawDB(func(db *gorm.DB) error {
+		err := store.ORM.Unscoped().RawDBWithAdvisoryLock(func(db *gorm.DB) error {
```

Comment:
My spidey senses are tingling due to the fact that:

1. Advisory locks are scoped to connections
2. Connections are opaquely pooled by the stdlib SQL package

...which means that this could fail nondeterministically